packages/gatsby/src/redux/reducers/set-request-headers.ts

```diff
@@ -0,0 +1,20 @@
+import { IGatsbyState, ISetDomainRequestHeaders } from "../types"
+
+/**
+ * Takes in a domain and headers for that domain, from the setRequestHeaders action, and stores them in a Map to be accessed when making requests.
+ */
+export const setRequestHeadersReducer = (
+  state: IGatsbyState["requestHeaders"] = new Map(),
+  action: ISetDomainRequestHeaders
+): IGatsbyState["requestHeaders"] => {
+  switch (action.type) {
+    case `SET_REQUEST_HEADERS`: {
+      const { headers, domain } = action.payload
+
+      state.set(domain, headers)
```

Comment:
Should we handle multiple source plugins setting headers for the same domain?

Reply:
I thought about that too but ended up deciding it's likely not needed. We only have 1 place grabbing these headers and it needs to grab them from all plugins anyway. I could add some code here that merges headers if the same domain is added twice. I'm not sure that will ever happen though so it might be better to wait and see.

Reply:
That make all makes sense :+1: 

Just merging headers straight up wouldn't be great either in the case that two different source plugins needed to pass different values for the same header. (e.g. different authorization header values for the same domain)

Reply:
Yeah definitely, so we should probably leave it as-is. The likelihood of 2 different source plugins sourcing from the same domain but needing different headers is tiny